src/state/queries/post-thread.ts

```diff
@@ -0,0 +1,200 @@
+import {AppBskyFeedDefs, AppBskyFeedGetPostThread} from '@atproto/api'
+import {useQuery, QueryClient, useQueryClient} from '@tanstack/react-query'
+import {useSession} from '../session'
+import {RQKEY as POST_RQKEY, getCachedPost} from './post'
+import {ThreadViewPreference} from '../models/ui/preferences'
+
+export const RQKEY = (uri: string) => ['post-thread', uri]
```

Comment:
Not positive we'd really see an issue with this, but we might want to include the user did or something in this keyset so that switching users without a reload to clear memory never results in stale data being shown from cache.

Reply:
I assumed we'd drop all RQ cache on a session change

Reply:
Would be nice to have our React Query Keys strongly typed in one place so we can refer to them for future updates. A very simple factory model should work: https://tkdodo.eu/blog/effective-react-query-keys#use-query-key-factories

Reply:
I'm +1 to this, nice to have a quick reference to what query keys are out there

Reply:
Actually @ansh does it change your mind any if we have separate query hooks for each query like Paul has here? I.e. `usePostQuery` instead of `useQuery(RQKEY('uri'), () => {})`. So finding all usages is pretty easy either way.